Makefile.shared

```diff
@@ -245,15 +245,15 @@ link_dso.darwin:
 	SHLIB_SUFFIX=.dylib; \
 	ALLSYMSFLAGS=''; \
 	NOALLSYMSFLAGS=''; \
-	SHAREDFLAGS="$(CFLAGS) `echo $(SHARED_LDFLAGS) | sed s/dynamiclib/bundle/`"; \
+	SHAREDFLAGS="`echo $(SHARED_LDFLAGS) | sed s/dynamiclib/bundle/`"; \
 	$(LINK_SO_DSO)
 link_shlib.darwin:
 	@ $(CALC_VERSIONS); \
 	SHLIB=lib$(LIBNAME); \
 	SHLIB_SUFFIX=.dylib; \
 	ALLSYMSFLAGS='-all_load'; \
 	NOALLSYMSFLAGS=''; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS)"; \
+	SHAREDFLAGS="$(SHARED_LDFLAGS)"; \
```

Comment:
> Supposedly "system" linker is llvm-aware and knows what to do even if it finds llvm byte-code in object file on command line. "Supposedly" means that I personally haven't double-checked it.

Double-checked. MacOS X "system" linker is llvm-aware and is capable to recognizing llvm byte-code even if file is named .o and act accordingly. This is unlike Linux.